src/reducers/fileTree.tsx

```diff
@@ -157,6 +159,74 @@ export const buildFileTree = (version: Version): FileTree => {
   };
 };
 
+type GetRelativePathParams = {
+  _log: typeof log;
+  currentPath: string;
+  pathList: string[];
+  position: 'previous' | 'next';
+};
+
+export const getRelativePath = ({
+  _log = log,
+  currentPath,
+  pathList,
+  position,
+}: GetRelativePathParams): string | void => {
+  const currentIndex = pathList.indexOf(currentPath);
+  if (!currentIndex) {
+    _log.debug(`Cannot find ${currentPath} in pathList: ${pathList}`);
+    return undefined;
+  }
+  const newIndex =
+    position === 'previous' ? currentIndex - 1 : currentIndex + 1;
+  const validIndex = newIndex >= 0 && pathList.length >= newIndex;
+  if (!validIndex) {
+    _log.debug(`There is not a ${position} file wrt ${currentPath}`);
+  }
+  return validIndex ? pathList[newIndex] : undefined;
+};
+
+type GoToNextFileParams = {
+  _log: typeof log;
+  currentPath: string;
+};
+
+export const goToNextFile = ({
+  _log = log,
+  currentPath,
+}: GoToNextFileParams): ThunkActionCreator => {
+  return async (dispatch, getState) => {
+    const { fileTree: fileTreeState } = getState();
+    const { forVersionId, tree } = fileTreeState;
+    const { pathList } = tree;
+
+    if (!forVersionId || !pathList) {
+      _log.debug('Cannot navigate to next file with no version loaded');
+      return;
+    }
+
+    const nextPath = getRelativePath({
+      _log,
+      currentPath,
+      pathList,
+      position: 'next',
+    });
+
+    if (!nextPath) {
+      // There is no next file, so do nothing.
+      // TODO: I don't think we need to log in this case.
+      return;
+    }
+
+    dispatch(
```

Comment:
Were you implying on IRC that the existence of a sole dispatch suggests you don't need a thunk and that you could move all of this to the versions reducer? I don't think it would buy you much because you'd have to duplicate the tests for `updateSelectedPath` and that would be annoying. 

Also, I can see a need to dispatch more actions (as commented above) so I think you probably need a thunk. Moving the thunk to the versions reducer won't buy you much either.